Assets/_Application/Layers/LayerTypes/CartesianTiles/CartesianTileLayerGameObject.MaterialApplicator.cs

```diff
@@ -0,0 +1,86 @@
+using Netherlands3D.CartesianTiles;
+using Netherlands3D.LayerStyles;
+using Netherlands3D.LayerStyles.Expressions;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Netherlands3D.Twin.Layers.LayerTypes.CartesianTiles
+{
+    public partial class CartesianTileLayerGameObject
+    {
+        internal class CartesianTileBinaryMeshLayerMaterialApplicator : IMaterialApplicatorAdapter
+        {
+            public List<int> MaterialIndices => materialIndices;
+            private List<int> materialIndices = new List<int>();
+
+            private readonly CartesianTileLayerGameObject layer;            
+
+            public CartesianTileBinaryMeshLayerMaterialApplicator(CartesianTileLayerGameObject layer)
+            {
+                this.layer = layer;
+                BinaryMeshLayer binaryMeshLayer = layer.layer as BinaryMeshLayer;
+                List<Material> materials = binaryMeshLayer.DefaultMaterialList;
+                int index = 0;
+                foreach (Material material in materials)
+                {                    
+                    layer.CreateFeature(material); //one feature per default shared material
+
+                    //populate the layerstyles based on available materials if not present in data yet
+                    LayerStyle style = new LayerStyle(material.name);
+                    StylingRule stylingRule = new StylingRule(material.name, new MatchMaterialNameExpression(material.name));
```

Comment:
Dit kan ervoor zorgen dat er duplicate StylingRules aangemaakt kunnen worden als je een project opslaat en inlaad. 

De StylingRules worden namelijk opgeslagen in de LayerData en zijn bedoeld om aan te maken zodra nodig -dus in dit geval zelfs als de LayerData voor het eerst wordt aangemaakt-.

Vandaar ook het idee dat je een LayerFeature per materiaal aanmaakt, en er dan een expressie kan komen die in de LayerFeature zijn attributes kijkt wat de naam van de material property is en op basis daarvan kan matchen